app/models/notification.rb

```diff
@@ -79,7 +87,47 @@ def unarchive_if_updated
   def update_from_api_response(api_response, unarchive: false)
     attrs = Notification.attributes_from_api_response(api_response)
     self.attributes = attrs
+    update_subject
     unarchive_if_updated if unarchive
     save(touch: false) if changed?
   end
+
+  private
+
+  def download_subject
+    user.github_client.get(subject_url)
+  end
+
+  def update_subject
+    return unless Octobox.config.fetch_subject
+    if subject
+      # skip syncing if the notification was updated around the same time as subject
+      return if updated_at - subject.updated_at < 2.seconds
+
+      case subject_type
+      when 'Issue', 'PullRequest'
+        remote_subject = download_subject
+        subject.state = remote_subject.merged_at.present? ? 'merged' : remote_subject.state
+        subject.save(touch: false) if subject.changed?
+      end
+    else
+      case subject_type
+      when 'Issue', 'PullRequest'
+        remote_subject = download_subject
+        create_subject({
```

Comment:
I did start with a `find_or_create` but the logic so different based on if it's a existing subject or one we need to download from fresh that it was more hassle than it's worth.

The nice thing about using `subject_url` as the foreign key is that every users notification for a given issue/pr/release will reference exactly the same subject, so we'll often have an existing subject in the db which we'll only conditionally need to check to see if it's updated based on that 2 second updated_at comparison.